addons/mail_group/controllers/main.py

```diff
@@ -0,0 +1,266 @@
+            })
+
+        return groups
+
+    @http.route('/groups', type='http', auth='public', sitemap=True, website=True)
+    def groups(self, email='', **kw):
+        """View of the group lists. Allow the users to subscribe and unsubscribe."""
+        mail_groups = request.env['mail.group'].sudo().search([])
+
+        # Get statistics about the number of messages per month
+        month_date = datetime.today() - relativedelta.relativedelta(months=1)
+        messages_data = request.env['mail.group.message'].read_group([
+            ('create_date', '>=', fields.Datetime.to_string(month_date)),
+            ('moderation_status', '=', 'accepted'),
+        ], ['mail_group_id'], ['mail_group_id'])
+
+        # Dictionary { mail_discusison_id: number_of_message_per_month }
+        messages_data = {
+            message['mail_group_id'][0]: message['mail_group_id_count']
+            for message in messages_data
+        }
+
+        if self._is_user_logged():
+            # Force the email if the user is logged
+            email = request.env.user.email
+
+        # List of group in which the current email is member
+        email_normalized = tools.email_normalize(email)
+        members_data = request.env['mail.group.member'].sudo().search([('email_normalized', '=', email_normalized)])
+        members_data = {member.mail_group_id.id: True for member in members_data}
+
+        return request.render('mail_group.mail_groups', {
+            'mail_groups': mail_groups,
+            'messages_data': messages_data,
+            'members_data': members_data,
+            'email': email,
+        })
+
+    @http.route('/groups/subscription', type='json', auth='public', website=True)
+    def groups_subscription(self, group_id=0, action='subscribe', email=None, **kw):
+        """Subscribe the current logged user or the given email address to the mailing list.
+
+        If the user is logged, the action is automatically done.
+
+        But if the user is not logged (public user) an email will be send with a token
+        to confirm the action.
+
+        :param group_id: Id of the group
+        :param action: Action to perform (subscribe or unsubscribe)
+        :param email: Email to add in the member list
+        :return:
+            'done'
+                if the action has been done immediately
+            'email_sent'
+                if we send a confirmation email
+            'is_already_member'
+                if we try to subscribe but we are already member
+            'is_not_member'
+                if we try to unsubscribe but we are not member
+        """
+        if action not in ('subscribe', 'unsubscribe'):
+            raise werkzeug.exceptions.NotFound()
+
+        group = request.env['mail.group'].browse(int(group_id)).exists().sudo()
+        if not group:
+            raise werkzeug.exceptions.NotFound()
+
+        is_member = request.env['mail.group.member'].sudo().search_count([
```

Comment:
Would check for user's partner if logged, otherwise fallback on email. Purpose would be to somehow synchronize user's email/partner if possible.

Reply:
But as we already force the synchronization with the compute is it really necessary ?

Reply:
Edit: now use `_get_member` method which care about the partner